BuildMacOSUniversalBinary.py

```diff
@@ -0,0 +1,298 @@
+# config options for several reasons:
+# 1) Adding new architectures will generally require more code changes
+# 2) Single architecture builds should utilize the normal generated cmake
+#    project files rather than this wrapper script
+
+ARCHITECTURES = ["x86_64", "arm64"]
+
+
+def parse_args(conf=DEFAULT_CONFIG):
+    """
+    Parses the command line arguments into a config dictionary.
+    """
+
+    parser = argparse.ArgumentParser(
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+
+    parser.add_argument(
+        "--target",
+        help="Build target in generated project files",
+        default=conf["build_target"],
+        dest="build_target")
+    parser.add_argument(
+        "-G",
+        help="CMake Generator to use for creating project files",
+        default=conf["generator"],
+        dest="generator")
+    parser.add_argument(
+        "--build_type",
+        help="CMake build type [Debug, Release, RelWithDebInfo, MinSizeRel]",
+        default=conf["build_type"],
+        dest="build_type")
+    parser.add_argument(
+        "--dst_app",
+        help="Directory where universal binary will be stored",
+        default=conf["dst_app"])
+
+    parser.add_argument(
+        "--entitlements",
+        help="Path to .entitlements file for code signing",
+        default=conf["entitlements"])
+
+    parser.add_argument(
+        "--codesign",
+        help="Code signing identity to use to sign the applications",
+        default=conf["codesign_identity"],
+        dest="codesign_identity")
+
+    for arch in ARCHITECTURES:
+        parser.add_argument(
+             f"--{arch}_cmake_prefix",
+             help="Folder for cmake to search for packages",
+             default=conf[arch+"_cmake_prefix"],
+             dest=arch+"_cmake_prefix")
+
+        parser.add_argument(
+             f"--{arch}_qt5_path",
+             help=f"Install path for {arch} qt5 libraries",
+             default=conf[arch+"_qt5_path"])
+
+        parser.add_argument(
+             f"--{arch}_mac_os_deployment_target",
+             help=f"Deployment architecture for {arch} slice",
+             default=conf[arch+"_mac_os_deployment_target"])
+
+    return vars(parser.parse_args())
+
+
+def lipo(path0, path1, dst):
+    if subprocess.call(["lipo", "-create", "-output", dst, path0, path1]) != 0:
+        print(f"WARNING: {path0} and {path1} cannot be lipo'd")
+
+        shutil.copy(path0, dst)
+
+
+def recursive_merge_binaries(src0, src1, dst):
+    """
+    Merges two build trees together for different architectures into a single
+    universal binary.
+
+    The rules for merging are:
+
+    1) Files that exist in either src tree are copied into the dst tree
+    2) Files that exist in both trees and are identical are copied over
+       unmodified
+    3) Files that exist in both trees and are non-identical are lipo'd
+    4) Symlinks are created in the destination tree to mirror the hierarchy in
+       the source trees
```

Comment:
I think it is worth checking just in case. I added some checks to detect this and throw an exception (since you can't build a valid universal binary if this occurs.